_posts/news/2017-11-29-announcing-scalajs-1.0.0-M2.md

```diff
@@ -0,0 +1,435 @@
+category: news
+tags: [releases]
+permalink: /news/2017/11/29/announcing-scalajs-1.0.0-M2/
+---
+
+
+We are very excited to announce the release of Scala.js 1.0.0-M2!
+
+This development release is mostly intended for testing purposes, and as a synchronization point with library authors so that they can start upgrading in preparation for the final release.
+
+As the change in "major" version number witnesses, this release is *not* binary compatible with 0.6.x, nor with 1.0.0-M1.
+Libraries need to be recompiled and republished using this milestone to be compatible.
+
+Moreover, this release is not entirely source compatible with 0.6.x either.
+We expect, however, that further milestones for 1.0.0 will stay backward source compatible with this milestone.
+
+**Attention:** We would like to bring your attention to the fact that Scala.js 1.0.0-M2 is the last release that will support:
+
+* sbt 0.13
+* building on JDK 6 and JDK 7
+* Scala 2.10
+
+Starting with Scala.js 1.0.0-M3, the support for those will be dropped, as those technologies are obsolete and going forward, would put an undue maintenance burden on our shoulders.
+
+<!--more-->
+
+Please report any issues [on GitHub](https://github.com/scala-js/scala-js/issues).
+
+## Preparations before upgrading from 0.6.x
+
+Before upgrading to 1.0.0-M2, **we strongly recommend that you upgrade to Scala.js 0.6.20 or later**, and address all deprecation warnings.
+Since Scala.js 1.0.0-M2 removes support for all the deprecated features in 0.6.x, it is easier to see the deprecation messages guiding you to the proper replacements.
+
+Additionally to the explicitly deprecated things, make sure to use `scalaJSLinkerConfig` instead of the following sbt settings:
+
+* `scalaJSSemantics`
+* `scalaJSModuleKind`
+* `scalaJSOutputMode`
+* `emitSourceMaps`
+* `relativeSourceMaps`
+* `scalaJSOutputWrapper`
+* `scalaJSOptimizerOptions`
+
+## Upgrade to 1.0.0-M2 from 0.6.20 or later
+
+As a first approximation, all you need to do is to update the version number in `project/plugins.sbt`:
+
+{% highlight scala %}
+addSbtPlugin("org.scala-js" % "sbt-scalajs" % "1.0.0-M2")
+{% endhighlight %}
+
+In addition, if you use some of the components that have been moved to separate repositories, you will need to add some more dependencies in `project/plugins.sbt`:
+
+If you use `crossProject`:
+
+* Add `addSbtPlugin("org.portable-scala" % "sbt-scalajs-crossproject" % "0.3.0")` in `project/plugins.sbt`
+
+If you use `jsDependencies` (or rely on the `jsDependencies` of your transitive dependencies):
+
+* Add `addSbtPlugin("org.scala-js" % "sbt-jsdependencies" % "1.0.0-M2")` in `project/plugins.sbt`
+* Add `.enablePlugins(JSDependenciesPlugin)` to Scala.js `project`s
+* Add `.jsConfigure(_.enablePlugins(JSDependenciesPlugin))` to `crossProject`s
+
+If you use the Node.js with jsdom environment:
+
+* Add `libraryDependencies += "org.scala-js" %% "scalajs-env-nodejs" % "1.0.0-M2"` in `project/plugins.sbt`
```

Comment:
Why is the "automatic selection with `jsDependencies`" part not here anymore?

Reply:
Because `RuntimeDOM` was deprecated in 0.6.20 with a deprecation message explaining to use `jsEnv := ...` explicitly already. So if people do upgrade to 0.6.20 and address deprecation warnings like we tell them to, by the time they get here this has already been taken care of.

Reply:
Fair enough. That's what I thought.